src/jlgen.jl

```diff
@@ -219,31 +44,31 @@ argument typed `tt`. If the `world` argument is specified, the look-up is static
 always return the same result. If the `world` argument is not specified, the look-up is
 dynamic and the returned method instance will automatically be invalidated when a relevant
 function is redefined.
+
+If the method is not found, a `MethodError` is thrown.
 """
 function methodinstance(ft::Type, tt::Type, world::Integer=tls_world_age())
     sig = typed_signature(ft, tt)
 
-    # look-up the method
-    if VERSION >= v"1.10.0-DEV.65"
-        meth = Base._which(sig; world).method
-    elseif VERSION >= v"1.7.0-DEV.435"
-        meth = Base._which(sig, world).method
+    # look-up the method (like `which`, but without throwing an error)
+    # TODO: use `CC.findsup`
+    meth = if v"1.8-beta2" <= VERSION < v"1.9-" || VERSION >= v"1.9.0-DEV.149"
+        # TODO: pass the correct `mt` argument (maybe fixes JuliaGPU/GPUCompiler.jl#208)
+        ccall(:jl_gf_invoke_lookup, Any, (Any, Any, UInt), sig, nothing, world)
```

Comment:
Note: this function is potentially going to be deprecated. The replacement is `jl_gf_invoke_lookup_worlds`.